source/main.cpp

```diff
@@ -365,160 +357,34 @@ WMMSettingsView::~WMMSettingsView()
-	fDemoView->StrokeRoundRect(demoRect, 2, 2);
-
-	InitDemo();
-}
-
-//---------------------------------------------------
-//	Draw demo
-//---------------------------------------------------
-void
-WMMSettingsView::DrawDemo()
+WMMSettingsView::Draw(BRect updateRect)
 {
-	float right = demoRect.right, bottom = demoRect.bottom;
-
-	if( demoCircle < Settings.circles)
-	{
-		float oldPen = fDemoView->PenSize();
-		fDemoView->SetPenSize(Settings.lwidth);
-		BRegion region = BRegion(BRect(1, 1, right - 1, bottom - 1));
-		fDemoView->ConstrainClippingRegion(&region);
-
-		fDemoView->SetHighColor(demoColor);
-		fDemoView->StrokeEllipse(BPoint(right / 2, bottom / 2), demoRadius, demoRadius);
-		demoRadius += ( Settings.lwidth / 2) + Settings.lspace;
-		demoColor.alpha += demoAlphaMod;
-		demoCircle++;
-
-		fDemoView->ConstrainClippingRegion(NULL);
-		fDemoView->SetPenSize(oldPen);
-	}
-	else
-	{
-		//demo background
-		fDemoView->SetHighColor(255, 255, 255, 255);
-		fDemoView->FillRoundRect(BRect(1, 1, right - 1, bottom - 1), 2, 2);
-		//SetHighColor( 0, 0, 0, 255);
-		//StrokeRoundRect( demoRect, 2, 2);
-
-		InitDemo();
-	}
+	fDemoView->Invalidate();
 }
 
 //---------------------------------------------------
 //	Message handler
 //---------------------------------------------------
 void
-WMMSettingsView::MessageReceived( BMessage *msg)
-{
-	// drag & drop
-	if( msg->WasDropped())
-	{
-		rgb_color*	color;
-		ssize_t		size;
-		if( (msg->FindData("RGBColor", B_RGB_COLOR_TYPE, (const void **)&color, &size) == B_OK))
-			Settings.color = *color;
-		return;
-	}
-
-	BRect rect = demoRect;
-	rect.left++;
-	rect.top++;
-	rect.right--;
-	rect.bottom--;
-
-	switch( msg->what)
-	{
-		case WMM_MSG_SET_RADIUS:
-		{
-			msg->FindInt32( "be:value", &Settings.radius);
-			fDemoView->Invalidate(rect);
-			InitDemo();
-			break;
-		}
-		case WMM_MSG_SET_CIRCLES:
-		{
-			msg->FindInt32( "be:value", &Settings.circles);
-			fDemoView->Invalidate(rect);
-			InitDemo();
-			break;
-		}
-		case WMM_MSG_SET_LWIDTH:
-		{
-			msg->FindInt32( "be:value", &Settings.lwidth);
-			fDemoView->Invalidate(rect);
-			InitDemo();
-			break;
-		}
-		case WMM_MSG_SET_LSPACE:
-		{
-			msg->FindInt32( "be:value", &Settings.lspace);
-			fDemoView->Invalidate(rect);
-			InitDemo();
-			break;
-		}
-		case WMM_MSG_SET_PULSE:
-		{
-			msg->FindInt32( "be:value", &Settings.pulse);
-			Settings.pulse = Settings.pulse * 100000;
-			Window()->SetPulseRate( Settings.pulse);
-			fDemoView->Invalidate(rect);
-			InitDemo();
-			break;
-		}
-		default:
-			BView::MessageReceived( msg);
-	}
-}
-
-//---------------------------------------------------
-//	Animate demo circles
-//---------------------------------------------------
-void
-WMMSettingsView::Pulse()
-{
-	DrawDemo();
-}
-
-//---------------------------------------------------
-//	Restart demo
-//---------------------------------------------------
-void
-WMMSettingsView::InitDemo()
+WMMSettingsView::MessageReceived(BMessage *msg)
```

Comment:
I left it empty because

void WMMSettingsView::MessageReceived(BMessage *msg)
{
	BView::MessageReceived(msg);
}

caused the follwoing error/warning

BHandler w>Where's my Mouse settings?: MessageReceived() couldn't understand the message:
BMessage('PSTE') {
        text/plain = (type = 'MIME')(size = 7)
        RGBColor = (type = 'RGBC')(size = 4)
        buttons = int32(0x1 or 1)
        _drop_point_ = BPoint(x:996, y:852)
        _drop_offset_ = BPoint(x:9, y:9)
}

when dropping something on the main view. (same thing when I removed the function)

What's the correct way to ignore and pass on all messages received by the main view, without causing the error/warning?